modules/widgets/social-icons.php

```diff
@@ -0,0 +1,500 @@
+							esc_attr( $args['url-icon-name'] ),
+							esc_attr__( 'Account URL', 'jetpack' ),
+							esc_url( $args['url-value'] )
+						);
+					?>
+				</p>
+
+				<p class="jetpack-widget-social-icons-remove-item">
+					<a class="jetpack-widget-social-icons-remove-item-button" href="javascript:;">
+						<?php esc_html_e( 'Remove', 'jetpack' ); ?>
+					</a>
+				</p>
+			</div>
+		</div>
+
+		<?php
+	}
+} // Jetpack_Widget_Social_Icons
+
+/**
+ * Register and load the widget.
+ *
+ * @access public
+ * @return void
+ */
+function jetpack_widget_social_icons_load() {
+	register_widget( 'Jetpack_Widget_Social_Icons' );
+}
+add_action( 'widgets_init', 'jetpack_widget_social_icons_load' );
+
+if ( ! function_exists( 'jetpack_social_menu_get_svg' ) ) :
+/**
+ * Return SVG markup.
+ *
+ * @param array $args {
+ *     Parameters needed to display an SVG.
+ *
+ *     @type string $icon  Required SVG icon filename.
+ * }
+ * @return string SVG markup.
+ */
+function jetpack_social_menu_get_svg( $args = array() ) {
+	// Make sure $args are an array.
+	if ( empty( $args ) ) {
+		return esc_html__( 'Please define default parameters in the form of an array.', 'jetpack' );
+	}
+
+	// Set defaults.
+	$defaults = array(
+		'icon'     => '',
+		'fallback' => false,
+	);
+
+	// Parse args.
+	$args = wp_parse_args( $args, $defaults );
+
+	// Define an icon.
+	if ( false === array_key_exists( 'icon', $args ) ) {
+		return esc_html__( 'Please define an SVG icon filename.', 'jetpack' );
+	}
+
+	// Set aria hidden.
+	$aria_hidden = ' aria-hidden="true"';
+
+	// Begin SVG markup.
+	$svg = '<svg class="icon icon-' . esc_attr( $args['icon'] ) . '"' . $aria_hidden . ' role="img">';
+
+	/*
+	 * Display the icon.
+	 *
+	 * The whitespace around `<use>` is intentional - it is a work around to a keyboard navigation bug in Safari 10.
+	 *
+	 * See https://core.trac.wordpress.org/ticket/38387.
+	 */
+	$svg .= ' <use href="#icon-' . esc_html( $args['icon'] ) . '" xlink:href="#icon-' . esc_html( $args['icon'] ) . '"></use> ';
+
+	// Add some markup to use as a fallback for browsers that do not support SVGs.
+	if ( $args['fallback'] ) {
+		$svg .= '<span class="svg-fallback icon-' . esc_attr( $args['icon'] ) . '"></span>';
+	}
+
+	$svg .= '</svg>';
+
+	return $svg;
+}
+endif;
+
+if ( ! function_exists( 'jetpack_social_menu_social_links_icons' ) ) :
+/**
+ * Returns an array of supported social links (URL and icon name).
+ *
+ * @return array $social_links_icons
+ */
+function jetpack_social_menu_social_links_icons() {
+	// Supported social links icons.
+	$social_links_icons = array(
+		'500px.com'       => '500px',
+		'amazon.cn'       => 'amazon',
+		'amazon.in'       => 'amazon',
+		'amazon.fr'       => 'amazon',
+		'amazon.de'       => 'amazon',
+		'amazon.it'       => 'amazon',
+		'amazon.nl'       => 'amazon',
+		'amazon.es'       => 'amazon',
+		'amazon.co'       => 'amazon',
+		'amazon.ca'       => 'amazon',
+		'amazon.com'      => 'amazon',
```

Comment:
Also, with regex matching we could make sure that `amazon.com` (as an example) is the tld, instead of a subdomain like `amazon.com.lolhax.io` which substr would consider to be amazon.

Not a huge security as it's just displaying an icon that could be easily mimicked via a text widget, but potentially a bit awkward.